frontend/wallet/src/main/DaemonProcess.re

```diff
@@ -27,70 +28,77 @@ let codaCommand = (~port, ~extraArgs) => {
         |],
       ),
     env:
-      Some(
-        Js.Dict.fromList([
-          (
-            "CODA_KADEMLIA_PATH",
-            ProjectRoot.resource ^/ codaPath ^/ "kademlia",
-          ),
-        ]),
+      Js.Dict.fromArray(
+        Js.Array.concat(
+          [|
+            (
+              "CODA_KADEMLIA_PATH",
+              ProjectRoot.resource ^/ codaPath ^/ "kademlia",
+            ),
+          |],
+          Js.Dict.entries(ChildProcess.Process.env),
+        ),
       ),
+    logfileName: "daemon",
   };
 };
 
 module Process = {
   let start = (command: Command.t) => {
-    let {Command.executable, args} = command;
-    print_endline({j|Starting $executable with $args|j});
+    let {Command.executable, args, logfileName} = command;
+    print_endline(
+      {j|Starting $executable with $args. Logging to `$logfileName.log`|j},
+    );
+
+    /*
+     child_process.spawn communicates with processes using sockets.
+     When the proposer process starts up in the daemon, stdout and stderr
+     are accessed using a helper function from async_unix:
+     https://github.com/janestreet/async_unix/blob/ff13d69c96f4b857737263910b3f6b08311b5dfc/src/writer0.ml#L1465
+     This causes async_unix to attempt to infer the type of stdout/err
+     using and check if it has SO_ACCEPTCONN set using getsockopt.
+     This flag is not supported on Mac, so it crashes with "Protocol not available"
+
+     As a workaround, we have to redirect all output to a file.
+
+     If this issue is ever resolved or we stop using async_unix, this code can be simplified:
+     https://github.com/janestreet/async_unix/issues/15
+     */
 
+    let log = Fs.openSync(command.logfileName ++ ".log", "w");
     let p =
-      switch (command.env) {
-      | None => ChildProcess.spawn(command.executable, command.args)
-      | Some(env) =>
-        ChildProcess.spawnWithEnv(
-          command.executable,
-          command.args,
-          {"env": env},
-        )
-      };
+      ChildProcess.spawn(
+        command.executable,
+        command.args,
+        {
+          "env": command.env,
+          "stdio":
+            ChildProcess.makeIOTriple(`Ignore, `Stream(log), `Stream(log)),
```

Comment:
would it make sense to write to a different log file onError?  Also, what happens to log file on restart, does it get rewritten since we use the same filename next time?  Would it benefit us in debugging to add timestamp to the .log file name?

Reply:
I figured we would revisit this when we decide what we want to do with logs, for now it's mostly just to unblock us. For now, I think this is easiest to work with as we will be frequently restarting the daemon for debugging etc and I don't want to create a lot of random files

Reply:
sg, makes sense